Dockerfile

```diff
@@ -43,8 +43,9 @@ RUN git clone -b $LIBCO_VER https://github.com/canonical/libco.git && \
 
 # --- Build raft
 
-RUN git clone -b $RAFT_VER https://github.com/canonical/raft.git && \
+RUN git clone https://github.com/canonical/raft.git && \
     cd raft && \
+    git checkout $RAFT_VER && \
```

Comment:
@erikwilson @galal-hussein is there a reason to favor clone-then-checkout-ref as opposed to just specifying the ref at clone time?

Reply:
using `-b` only works with branches.  or do you mean some other flags/process?

Reply:
oh, wow. for some reason i had it stuck in my head that it was a ref-ish aka branch/tag/commit